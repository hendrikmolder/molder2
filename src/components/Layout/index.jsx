import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import Helmet from 'react-helmet'
import { Container } from 'semantic-ui-react'

import SiteNavigation from '../SiteNavigation'
import PageMetadata from '../PageMetadata'

import 'prismjs/themes/prism-solarizedlight.css'
import 'prismjs/plugins/line-numbers/prism-line-numbers.css'
import 'semantic-ui-css/semantic.min.css' /* Import Semantic-UI CSS */

const MENU_ITEMS = [
    { name: "linkedin", href: "https://linkedin.com/in/hendrikmolder" },
    { name: "music", href: "/music" },
    { name: "blog", href: "/blog" },
]

const StyledContainer = styled(Container)`
    min-height: 100%;
    margin: 60px auto 100px;

    img, iframe, pre, code {
        width: 100%;
    }

    a {
        text-decoration: underline;
        color: #24292E;
    }
`

const MetaContainer = styled.div`
    margin: 0 0 24px;
`

const SubTitle = styled.h2`
    color: rgb(132, 132, 132);
    font-size: 20px;
    font-weight: 300;
    margin: 0 0 36px;
`

const Page = (props) => {
    const { children, title, showTitle, subTitle, meta, text } = props
    return (
        <React.Fragment>
            <SiteNavigation items={MENU_ITEMS} />
            <StyledContainer text={text && text}>
                <Helmet>
                    <title>{ title ? `${title} | molder` : `molder` }</title>
                </Helmet>

                <MetaContainer>
                    { meta && <PageMetadata {...meta} /> }
                </MetaContainer>
                { showTitle !== "false" ? title && <h1>{title}</h1> : null}
                { subTitle && <SubTitle>{subTitle}</SubTitle>}
                { children }
            </StyledContainer>
        </React.Fragment>
    )
}

Page.propTypes = {
    children: PropTypes.oneOfType([
        PropTypes.array,
        PropTypes.object
    ]).isRequired,
    title: PropTypes.string,
    showTitle: PropTypes.bool,
    subTitle: PropTypes.string,
    text: PropTypes.bool,
    meta: PropTypes.object
}

export default Page