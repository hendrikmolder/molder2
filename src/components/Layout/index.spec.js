import React from 'react'
import renderer from 'react-test-renderer'

import Layout from './index'

const TEST_TITLE = 'Test'
const TEST_SUBTITLE = 'test'

describe('Layout', () => {
    it('renders correctly', () => {
        const tree = renderer
            .create(<Layout title={TEST_TITLE} subTitle={TEST_SUBTITLE}><p>test</p></Layout>)
            .toJSON()
        expect(tree).toMatchSnapshot()
    })
})