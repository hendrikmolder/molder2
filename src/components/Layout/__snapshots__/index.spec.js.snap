// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout renders correctly with title 1`] = `
Array [
  <div
    className="sc-bdVaJa ekuuOW"
  >
    <a
      href="/"
    >
      home
    </a>
    <a
      href="https://linkedin.com/in/hendrikmolder"
    >
      linkedin
    </a>
    <a
      href="/music"
      partiallyActive={true}
    >
      music
    </a>
    <a
      href="/blog"
      partiallyActive={true}
    >
      posts
    </a>
  </div>,
  <div
    className="ui container sc-ifAKCX OLGY"
  >
    <h1>
      Test
    </h1>
    <h2
      className="sc-bZQynM ffMfhA"
    >
      test
    </h2>
    <p>
      test
    </p>
  </div>,
]
`;

exports[`Layout renders correctly without title 1`] = `
Array [
  <div
    className="sc-bdVaJa ekuuOW"
  >
    <a
      href="/"
    >
      home
    </a>
    <a
      href="https://linkedin.com/in/hendrikmolder"
    >
      linkedin
    </a>
    <a
      href="/music"
      partiallyActive={true}
    >
      music
    </a>
    <a
      href="/blog"
      partiallyActive={true}
    >
      posts
    </a>
  </div>,
  <div
    className="ui container sc-ifAKCX OLGY"
  >
    <p>
      test
    </p>
  </div>,
]
`;
